File: app/src/main/java/com/anokmik/databinding/model/ObservableString.java
package com.anokmik.databinding.model;

import android.databinding.BaseObservable;

public class ObservableString extends BaseObservable {

    String value;

    public ObservableString() {

    }

    public ObservableString(String value) {
        this.value = value;
    }

    public String get() {
        return (value != null) ? value : "";
    }

    public void set(String value) {
        this.value = value;
        notifyChange();
    }

    public int getLength() {
        return (value != null) ? value.length() : 0;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: app/src/main/kotlin/com/anokmik/databinding/model/ObservableString.kt
package com.anokmik.databinding.model

import android.databinding.BaseObservable

class ObservableString : BaseObservable() {

    var value: String? = null

    fun get(): String {
        return value ?: ""
    }

    fun set(value: String) {
        this.value = value
        notifyChange()
    }

    fun length(): Int {
        return value?.length ?: 0
    }

    override fun toString(): String {
        return get()
    }

}